go/runtime/push.go

```diff
@@ -96,11 +107,18 @@ func (f *FlowConsumer) Push(stream capture.Runtime_PushServer) error {
 
 		for {
 			var docs, checkpoint, err = capture.ReadPushCheckpoint(stream, maxPushByteSize)
+			logrus.Info(fmt.Sprintf("Got %d docs, checkpoint: %v, error: %v", len(docs), checkpoint, err))
+			var ackCh = client.NewAsyncOperation()
 			if err != nil {
 				return fmt.Errorf("reading push checkpoint: %w", err)
 			} else if err = push.Push(docs, checkpoint, ackCh); err != nil {
 				return fmt.Errorf("staging push for capture: %w", err)
 			}
+			// ... Is this the right way to do this?
```

Comment:
I spent some time trying to answer this question, and I don't think this is _wrong_, but I think it can probably be simplified. I _think_ a simpler approach would be to get rid of `ackErrs` entirely and create the `AsyncOperation` up on line 101. Then await the completion of the `AsyncOperation` down on line 137 using `<-asyncOp.Done():` instead of awaiting it in this separate goroutine. Basically, I don't see a benefit to indirecting through `ackErrs`.